substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -949,20 +984,53 @@ impl<'a, E: Ext, M: ?Sized + Memory<E::T>> Runtime<'a, E, M> {
 		} else {
 			self.ext.get_storage(&key)
 		};
+
 		if let Some(value) = outcome {
 			self.adjust_gas(charged, costs(value.len() as u32));
-			self.write_sandbox_output(
-				memory,
-				out_ptr,
-				out_len_ptr,
-				&value,
-				false,
-				already_charged,
-			)?;
-			Ok(ReturnErrorCode::Success)
+
+			match read_mode {
+				StorageReadMode::EthSemantics => {
+					let mut fixed_output = [0u8; 32];
+					let len = core::cmp::min(value.len(), fixed_output.len());
```

Comment:
nit can't we just write 
```suggestion
					let len = value.len().min(fixed_output.len());
```

Reply:
looking..